core/src/replay_stage.rs

```diff
@@ -3074,6 +3074,37 @@ impl ReplayStage {
 
                                 tower.vote_state.root_slot = bank_vote_state.root_slot;
                                 tower.vote_state.votes = bank_vote_state.votes;
+
+                                let last_voted_slot = tower.vote_state.last_voted_slot().unwrap_or(
```

Comment:
doesn't this have to be `Some` because we checked 

`if bank_vote_state.last_voted_slot()
                                > tower.vote_state.last_voted_slot()` 

above, so there must be at least some vote in `bank_vote_state.last_voted_slot()`

Reply:
not necessarily, see here:
```
bank_vote_state
    .votes
    .retain(|lockout| lockout.slot() > local_root);
```